Tests/PcppTestUtilities/PacketFactory.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "RawPacket.h"
+#include "Resources.h"
+
+namespace pcpp_tests
+{
+	namespace utils
+	{
+		class PacketFactory
+		{
+		public:
+			/// @brief The time used for creating packets in the factory.
+			timespec factoryTime;
+			/// @brief The default link layer type for packets created by this factory.
+			pcpp::LinkLayerType defaultLinkType = pcpp::LinkLayerType::LINKTYPE_ETHERNET;
+
+			/// @brief Creates a new PacketFactory instance with the current time as factoryTime.
+			PacketFactory();
+			explicit PacketFactory(timespec time);
+			explicit PacketFactory(timeval time);
+			explicit PacketFactory(pcpp::LinkLayerType linkType);
+
+			PacketFactory withTime(timespec time);
+			PacketFactory withTime(timeval time);
+			PacketFactory withLinkType(pcpp::LinkLayerType linkType);
```

Comment:
Some public methods, classes and enums in this PR have doxygen documentation and some don't. I'm ok with not having any doxygen documentation because these are internal entities, but let's be consistent?